### ISSUE TITLE
Behavior at jump discontinuities

### ISSUE THREAD

**Issue by hazelnb** [stevenpetryk/mafs#133]
Consider the plot `<Plot.OfX y={(x: number) => math.mod(x, 1)}/>`. This of course has a number of jump discontinuities, but by default mafs joins discontinuities with a vertical line. Personally, I find this behavior unexpected and undesirable and would like to at least have the option to configure the discontinutiy behavior. Something like Mathematica's `Exclusions` and `ExclusionsStyle` seem appropriate for the existing Mafs API. I.e., the user provides `Plot.OfX` with a parameter which tells it where the discontinuities are and how they should be drawn. 

To my mind, this should either be:
1.  an array of x values of discontinuities
2. a function `(x: number): boolean`

I think the "jump" style that I'm hoping for can be accomplished by modifying the `onPoint` callback in `sampleParametric` to output an M command rather than an L command if the t value of the point is a discontiuity. 

Let me know what you think! I love the library, and I'm using it to make companion interactive visualizations for my dissertation :)

**Comment by stevenpetryk** [comment]
Totally agreed. I think the function approach might be required for cases where there are infinite discontinuities, but instead of accepting a number and returning a boolean, it should accept a range and return a list (with the option to pass a static list, no function).

Mafs could use those values and:

- Evaluate continuity (or the user can explicitly specify)
- Ensure additional points are plotted near these important points (whether continuous or not)

In other words, a way to indicate important regions to plot.

**Comment by stevenpetryk** [comment]
Agreed also on the approach for jump discontinuities. 

For holes, the best approach would probably just be a two `<circle>`s: one to mask off the function, and one to show the classic circular outline. 

This means adding a mask to some functions which may have an effect on perf, but I doubt it.

**Comment by stevenpetryk** [comment]
Got a nice API for holes I think (not jumps yet, because the `sample` function scares me after not working with it for over a year).

<img width="762" alt="" src="https://github.com/stevenpetryk/mafs/assets/1724000/eced366c-3310-4e8d-9fb9-8635bb4a6c80">

**Comment by stevenpetryk** [comment]
<img width="116" alt="CleanShot 2024-01-18 at 13 20 24@2x" src="https://github.com/stevenpetryk/mafs/assets/1724000/872fc724-5949-49dd-955c-66efcc532526">

I think JavaScript's stupidity will prevent a lot of ergonomics here. But I'm okay with devs having to do the math to tell Mafs where to display these indicators.

**Comment by hazelnb** [comment]
Wow thanks for starting working on this so quickly, this looks great! I think your modifications for the "jump" api make sense, and if you're open to doing code review I would be happy to take a crack at implementing it. 

As a mathematician, I swear Javascript makes some truly odd choices for built in operators that have caused some surprising and sneaky bugs for my viz. I have certainly not come to expect ergonomics haha.
